models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -477,11 +477,7 @@ public function filterMultipleAssignments($data, $container, $params)
                 } elseif (!isset($relationItems[$elementHash])) {
                     $relationItems[$elementHash] = $item;
                 } else {
-                    @trigger_error(
-                            'Passing relations multiple times is deprecated since version 6.5.2 and will throw exception in 7.0.0, tried to assign ' . $elementHash
-                            . ' multiple times in field' . $fieldName . ' of object id: ' . $objectId,
-                            E_USER_DEPRECATED
-                        );
+                    throw new \Exception(sprintf('Passing relations multiple times is not allowed: %s in field %s of object id: %s', $elementHash, $fieldName, $objectId));
```

Comment:
@brusch Is there a plan for custom exceptions in Pimcore 7? It would be good to have different exception classes for catching

Reply:
@blankse yes, we can think about that where it makes sense 👍